src/DssSpell.sol

```diff
@@ -49,82 +121,192 @@ contract DssSpellAction is DssAction {
     //    https://ipfs.io/ipfs/QmVp4mhhbwWGTfbh2BzwQB9eiBrQBKiqcPRZCaAxNUaar6
     //
 
-    // --- Rates ---
-    // uint256 internal constant SEVEN_PT_FIVE_PERCENT_RATE = 1000000002293273137447730714;
+    function actions() public override {
+        // ----------------- Compound v2 D3M Onboarding -----------------
+        // https://vote.makerdao.com/polling/QmWYfgY2#poll-detail
+        {
+            VatAbstract vat = VatAbstract(DssExecLib.vat());
+            SpotAbstract spot = SpotAbstract(DssExecLib.spotter());
+            address daiJoin = DssExecLib.daiJoin();
+            address dai = DssExecLib.dai();
+            address vow = DssExecLib.vow();
+            address end = DssExecLib.end();
 
-    // --- Math ---
-    uint256 constant internal WAD     = 10 ** 18;
-    uint256 constant internal MILLION = 10 ** 6;
+            // Sanity checks
+            require(D3MHubLike(D3M_HUB).vat() == address(vat), "Hub vat mismatch");
+            require(D3MHubLike(D3M_HUB).daiJoin() == daiJoin, "Hub daiJoin mismatch");
 
-    address internal immutable MIP21_LIQUIDATION_ORACLE = DssExecLib.getChangelogAddress("MIP21_LIQUIDATION_ORACLE");
+            require(D3MCompoundPoolLike(D3M_COMPOUND_POOL).ilk() == ILK, "Pool ilk mismatch");
+            require(D3MCompoundPoolLike(D3M_COMPOUND_POOL).vat() == address(vat), "Pool vat mismatch");
+            require(D3MCompoundPoolLike(D3M_COMPOUND_POOL).dai() == dai, "Pool dai mismatch");
 
-    function _updateDoc(bytes32 ilk, string memory doc) internal {
-        ( , address pip, uint48 tau, ) = RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).ilks(ilk);
-        require(pip != address(0), "DssSpell/nonexistent-rwa-ilk");
+            require(D3MOracleLike(D3M_ORACLE).vat() == address(vat), "Oracle vat mismatch");
+            require(D3MOracleLike(D3M_ORACLE).ilk() == ILK, "Oracle ilk mismatch");
 
-        // Init the RwaLiquidationOracle to reset the doc
-        RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).init(
-            ilk, // ilk to update
-            0,   // price ignored if init() has already been called
-            doc, // new legal document
-            tau  // old tau value
-        );
-    }
+            D3MHubLike(D3M_HUB).file(ILK, "pool", D3M_COMPOUND_POOL);
+            D3MHubLike(D3M_HUB).file(ILK, "plan", D3M_COMPOUND_PLAN);
+            D3MHubLike(D3M_HUB).file(ILK, "tau", 7 days);
```

Comment:
Where was this value defined?

Reply:
Ok, but I guess we need someone else to sign off in that value? Should be risk?

Reply:
I'll ping them.

Reply:
Discussed in the chat with Risk. 7 days is fine.